README.md

```diff
@@ -144,6 +144,14 @@ If you see this error during an update of the SAM-TRE stack:
 
 Solution: Terminate your EC2 instances named TerraformExecutionInstance. Then rerun the failed command.
 
+#### NOTE: Consider following the below steps if you are attempting to update an already in-use installation
+1. **Manually disable the Lambda subscription to the provisioning intake SQS queues.** This will allow messages to be sent to the queues by Service Catalog but prevent the EC2 instance(s) from starting new workflows that could be affected by shutting down the EC2 instances in following steps
+1. **Navigate to Step Functions console and confirm there are no ongoing workflow executions. If there are, wait until they are complete before proceeding to the next step.** Similar to the last step, this will help ensure no ongoing executions are impacted by shutting down the EC2 instances in the following steps.
+1. **Shut down the instances in the ASG for this installation.** This will allow the deployment to update the attached security groups.
```

Comment:
I think ASG will immediately launch a new instance with existing security group configuration if the existing instance is shut down.

We could probably do the next step first to let CFN do the security group update and then manually shut down the instance when the stack waits for the EC2 dependency?

If we want to avoid doing that hack, we could also potentially add a step before this to delete the existing security group first?

Reply:
I may be missing your point so please correct me if I am. I personally think it's better to prepare for the deployment to succeed rather than kick off the deployment and expect users to perform a manual action during the deployment. Is your point that if we kick off the deployment before shutting down the instance then we minimize the time it's shut down? That will be true but the effect will be the same if users choose to follow the first two steps. The EC2 instances won't be processing any work regardless after the state machine executions are drained so it shouldn't matter whether the instances are running or not.

Again, let me know if I'm misunderstanding your point here.

Reply:
I think this would not work since you do the manual shutdown before CFN creates the new security group. As a result, another EC2 instance with the old security group will be backed up by the ASG and if you would still have a dependency on the old security group by the time you kick off the deployment.

We should probably just have the customer kick off the CFN deployment, and then have them monitor the CFN events to watch for a `DELETE_FAILED` status for the `InstanceSecurityGroup` resource with message like `resource sg-xxxxxxxxxxxx has a dependent object (Service: AmazonEC2; Status Code: 400; Error Code: DependencyViolation; Request ID: xxxxxxxxxxxx; Proxy: null)`. And then go to EC2 console to shut down the instance.

Reply:
Ah I understand now; that sounds correct. I will update in the next revision.